newsrevenuehub/rev-engine
# NRH-284 - Add Support for Google Analytics v4 for Org

#### What's this PR do?

- Hub admins can add a value for `google_analytics_v4_id` to a revenue program
- Any public facing donation or thank you page for an org whose revenue program has `google_analytics_v4_id` set will fire page view events via Google Analytics v4 plugin.

#### How should this be manually tested?

- Add a Google Analytics v4 id (it will be prefixed with `G-`) to a revenue program
- Visit a live page belonging to that revenue program
- In the network tab of dev console, confirm that a pageview event gets sent via request to `https://www.google-analytics.com/g/collect`. **Note** the `/g/` in that URL. It is distinct from `https://www.google-analytics.com/j/collect`, which is Google's endpoint for v3 analytics page view events. We will also be firing a page view event to that endpoint for the Hub GA, but the relevant one for this PR is the one to `https://www.google-analytics.com/g/collect`.
- Confirm in Google analytics dashboard that page view event is firing as expected

#### Do any changes need to be made before deployment to staging? production? (adding environment variables, for example)?

No
